spatialmath/capsule.go

```diff
@@ -0,0 +1,346 @@
+package spatialmath
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+
+	"github.com/golang/geo/r3"
+	commonpb "go.viam.com/api/common/v1"
+
+	"go.viam.com/rdk/utils"
+)
+
+// capsule is a collision geometry that represents a capsule, it has a pose and a radius that fully define it.
+//
+// ....___________________
+// .../                   \
+// .x|  |-------O-------|  |x
+// ...\___________________/
+//
+// Length is the distance between the x's, or internal segment length + 2*radius.
```

Comment:
To be clear, this is not a big deal to me.

Reply:
I put quite a bit of thought into that choice and went back and forth. This is purely a user experience decision, not something that impacts collision checks at all.

I agree that specifying segment length rather than total length makes for a "cleaner" mathematical representation, for exactly the reasons you mentioned.

However, if we do that, we make it mildly more difficult for a user to actually use. Currently the values are ones that are easy to physically measure. Let's say we go to make a capsule representation of the internal geometries of a UR5 next week. We pull out a tape measure, say "this limb is 175mm long, and 34.3mm diameter, so 17.15mm radius" and we plug that in. We don't have to worry about simple arithmetic to compute the internal segment length, which isn't something that's easily physically measured.

The majority of capsule implementations I found in the wild similarly specify full length, at least at the user-facing interfaces.

Reply:
If you can't measure both aspects I'm not sure what the user is doing.